.github/workflows/release.yml

```diff
@@ -0,0 +1,182 @@
+name: Ion Python Release
+
+on:
+  push:
+    branches: [ master ]
+    tags:
+      - "v*.*.*"
```

Comment:
I know `ion-dotnet` uses this event for the workflow, but have you considered using the [`release`](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#release) event?

Reply:
Looked into the document, seems like **release event** covers **tag** situation and it's more focusing on release. I think it's a good idea to use release, I can also send a PR to change ion-dotnet later.

Reply:
Addressed in below commit.